main/src/library/Fixpoint/Ast/PredSym.flix

```diff
@@ -20,8 +20,17 @@ namespace Fixpoint/Ast {
     }
 
     instance Order[PredSym] {
-        pub def compare(a: PredSym, b: PredSym): Comparison = match (a, b) {
-            case (PredSym(s1), PredSym(s2)) => s1 <=> s2
+        pub def compare(a: PredSym, b: PredSym): Comparison = match a {
+            case PredSym(s1) => match b {
+                case PredSym(s2) => s1 <=> s2
+            }
+        }
+    }
+
+    instance Eq[PredSym] {
+        pub def eq(x: PredSym, y: PredSym): Bool = match x <=> y {
```

Comment:
This is probably not that great for performance. It would be better to use the underlying Eq instance .